src/render/filters.rs

```diff
@@ -164,6 +167,47 @@ impl ResolveFilter for CapfirstFilter {
     }
 }
 
+impl ResolveFilter for CenterFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        template: TemplateString<'t>,
+        context: &mut Context,
+    ) -> ResolveResult<'t, 'py> {
+        let left: usize;
+        let right: usize;
+        let arg_size: usize;
+        let content = match variable {
+            Some(content) => {
+                content.render(context)?.into_owned()
+            },
+            None => return Ok("".as_content()),
+        };
+        let arg = self
+            .argument
+            .resolve(py, template, context, ResolveFailures::Raise)?
+            .expect("missing argument in context should already have raised");
+        let arg_int = arg.to_bigint();
+        match arg_int {
+            Some(arg_int) => {
+                arg_size = arg_int.to_usize().unwrap();
```

Comment:
As noted above, I'd like us to return an error if the `arg` can't be turned into a `usize` - not panic, which `unwrap()` would do.